src/query/functions/src/scalars/bitmap.rs

```diff
@@ -144,6 +144,59 @@ pub fn register(registry: &mut FunctionRegistry) {
         ),
     );
 
+    registry.register_passthrough_nullable_3_arg::<BitmapType, UInt64Type, UInt64Type, BitmapType, _, _>(
+        "bitmap_subset_limit",
+        |_, _, _| FunctionDomain::MayThrow,
+        vectorize_with_builder_3_arg::<BitmapType, UInt64Type, UInt64Type, BitmapType>(
+            |b, range_start, limit, builder, ctx| match RoaringTreemap::deserialize_from(b) {
+                Ok(rb) => {
+                    if limit > rb.len() {
+                        let empty_rb = RoaringTreemap::new();
+                        empty_rb.serialize_into(&mut builder.data).unwrap();
+                    } else {
+                        let subset_start = rb.iter().position(|x| x == range_start).unwrap_or(0);
+                        let subset_bitmap = rb.iter().skip(subset_start).take(limit as usize);
+                        let subset_rb: RoaringTreemap = subset_bitmap.collect();
+                        subset_rb.serialize_into(&mut builder.data).unwrap();
+                    }
+                    builder.commit_row();
+                }
+                Err(e) => {
+                    ctx.set_error(builder.len(), e.to_string());
+                }
+            },
+        ),
+    );
+
+    registry.register_passthrough_nullable_3_arg::<BitmapType, UInt64Type, UInt64Type, BitmapType, _, _>(
+        "bitmap_subset_in_range",
+        |_, _, _| FunctionDomain::MayThrow,
+        vectorize_with_builder_3_arg::<BitmapType, UInt64Type, UInt64Type, BitmapType>(
+            |b, start, end, builder, ctx| {
+                match RoaringTreemap::deserialize_from(b) {
+                    Ok(rb) => {
+                        let subset_start = start;
+                        let subset_end = end;
+                        if end >= rb.len() || subset_start > subset_end {
+                            let empty_rb = RoaringTreemap::new();
+                            empty_rb.serialize_into(&mut builder.data).unwrap();
+                        } else {
+                            let subset_bitmap = rb
+                                .into_iter()
+                                .filter(|&x| x >= subset_start && x <= subset_end)
+                                .collect::<RoaringTreemap>();
+                            subset_bitmap.serialize_into(&mut builder.data).unwrap();
+                        }
```

Comment:
`rb.iter().filter(|x| x >= start && x < end)` is ok.

Reply:
So with these review , you examples will work ?

Reply:
yeah, you can test it in your local.

Reply:
@gitccl what a problem which I have , I can not build databend in my local because of RAM shortage :)